src/restic/backend/gs/gs.go

```diff
@@ -0,0 +1,461 @@
+package gs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"restic"
+	"strings"
+	"time"
+
+	"restic/backend"
+	"restic/errors"
+	"restic/debug"
+
+	"io/ioutil"
+
+	"golang.org/x/oauth2/google"
+	storage "google.golang.org/api/storage/v1"
+	"google.golang.org/api/googleapi"
+)
+
+// Backend stores data on an gs endpoint.
+type Backend struct {
+	service    *storage.Service
+	projectID  string
+	location   string
+	sem        *backend.Semaphore
+	bucketName string
+	prefix     string
+	backend.Layout
+}
+
+// make sure that *Backend implements backend.Backend
+var _ restic.Backend = &Backend{}
+
+const defaultLayout = "default"
+
+func getStorageService(jsonKeyPath string) (*storage.Service, error) {
+
+	raw, err := ioutil.ReadFile(jsonKeyPath)
+	if err != nil {
+		return nil, err
```

Comment:
Use `errors.Wrap(err, "ReadFile")` (which uses https://github.com/pkg/errors) here to wrap the error with a stack trace. The same goes for all other errors coming from outside restic.